packages/flutter/lib/src/material/date_picker_theme.dart

```diff
@@ -0,0 +1,930 @@
+  final MaterialStateProperty<Color?>? dayForegroundColor;
+
+  /// The color used to paint the background of the day labels in the grid
+  /// of the date picker.
+  final MaterialStateProperty<Color?>? dayBackgroundColor;
+
+  /// The highlight color that's typically used to indicate that
+  /// a day in the grid is focused, hovered, or pressed.
+  final MaterialStateProperty<Color?>? dayOverlayColor;
+
+  /// The color used to paint the [DatePickerDialog.currentDate] label in
+  /// the grid of the date picker.
+  ///
+  /// This will be used instead of the color provided in [dayStyle].
+  final MaterialStateProperty<Color?>? todayForegroundColor;
+
+  /// The color used to paint the background of the
+  /// [DatePickerDialog.currentDate] label in the grid of the date picker.
+  final MaterialStateProperty<Color?>? todayBackgroundColor;
+
+  /// The border used to paint the [DatePickerDialog.currentDate] label in
+  /// the grid of the date picker.
+  final BorderSide? todayBorder;
+
+  /// The text style used to paint each of the year entries in the year
+  /// selector of the date picker.
+  ///
+  /// The color of the [yearStyle] is not used directly, as
+  /// [yearForegroundColor] is used instead.
+  final TextStyle? yearStyle;
+
+  /// The color used to paint the year labels in the year selector of
+  /// the date picker.
+  ///
+  /// This will be used instead of the color provided in [yearStyle].
+  final MaterialStateProperty<Color?>? yearForegroundColor;
+
+  /// The color used to paint the background of the year labels in the
+  /// year selector of the of the date picker.
+  final MaterialStateProperty<Color?>? yearBackgroundColor;
+
+  /// The highlight color that's typically used to indicate that
+  /// a year in the year selector is focused, hovered, or pressed.
+  final MaterialStateProperty<Color?>? yearOverlayColor;
+
+  /// The elevation of the full screen [DateRangePickerDialog].
+  ///
+  /// {@macro flutter.material.material.elevation}
+  final double? rangePickerElevation;
+
+  /// The color of the shadow painted below a full screen
+  /// [DateRangePickerDialog].
+  ///
+  /// {@macro flutter.material.material.shadowColor}
+  final Color? rangePickerShadowColor;
+
+  /// The color of the surface tint overlay applied to the [backgroundColor]
+  /// of a full screen [DateRangePickerDialog]'s to indicate elevation.
+  ///
+  /// {@macro flutter.material.material.surfaceTintColor}
+  final Color? rangePickerSurfaceTintColor;
+
+  /// The overall shape of a full screen [DateRangePickerDialog].
+  ///
+  /// {@macro flutter.material.material.shape}
+  final ShapeBorder? rangePickerShape;
+
+  /// The full screen [DateRangePickerDialog] header's background fill color.
+  final Color? rangePickerHeaderBackgroundColor;
+
+  /// The color used for text labels and icons in the header of a full
+  /// screen [DateRangePickerDialog]
+  ///
+  /// This is used instead of any colors provided by
```

Comment:
So why do those exist?

Reply:
This kind of issue comes up all of the time with compound visual style properties like BorderSide and TextStyle. The motive for setting up TextStyle properties with a separate text color property is two common cases:
- Use the default TextStyle (in this case) but override the default text color. Don't make the user create a TextStyle that matches the default just to change the color.
- Override the default TextStyle but do not override the default text color. Similar to above and note that the default color for a TextTheme TextStyle is likely to be wrong.